python/ray/state.py

```diff
@@ -385,6 +385,66 @@ def client_table(self):
 
         return _parse_client_table(self.redis_client)
 
+    def _job_table(self, job_id):
+        """Fetch and parse the job table information for a single job ID.
+
+         Args:
+            job_id: A job ID to get information about.
+         Returns:
+            A dictionary with information about the job ID in question.
+        """
+        # Allow the argument to be either a DriverID or a hex string.
+        if not isinstance(job_id, ray.JobID):
+            job_id = ray.JobID(hex_to_binary(job_id))
```

Comment:
Also assert the type of `job_id` is str?

Reply:
fixed